common4j/src/main/com/microsoft/identity/common/java/controllers/ExceptionAdapter.java

```diff
@@ -436,6 +439,18 @@ public static ClientException clientExceptionFromException(@NonNull final Throwa
 
 
         if (e instanceof OutOfMemoryError) {
+            // Log status code and record exception in span
+            final Span currentSpan = SpanExtension.current();
+
+            // Attach the stack trace, to debug in telemetry later
+            currentSpan.setAttribute(
+                    AttributeName.out_of_memory_exception_stacktrace.name(),
+                    StringUtil.getStacktraceAsStringFromElementArray(e.getStackTrace())
+            );
+
+            com.microsoft.identity.common.java.logging.Logger.warn(
+                    TAG, "Received an out of memory error, stacktrace attached to span.");
```

Comment:
Do you also want to add spanId in this log so that you can look for the exact device and what caused that particular device to throw OOM error?

Reply:
Yeah we can add that